### DISCUSSION TITLE
So many integrations not working...

### DISCUSSION THREAD

**Discussion by andreagaspari** [woocommerce/woocommerce#48902] [Checkout Flow]
I would like to use the new checkout flow with Gutenberg, but at the moment there are so many incompatible plugins, like [Mailchimp for Woocommerce](https://it.wordpress.org/plugins/mailchimp-for-woocommerce/), [Payment Plugins for PayPal WooCommerce](https://wordpress.org/plugins/pymntpl-paypal-woocommerce/), etc...
It would be great if you could ensure backwards compatibility with the old checkout flow so you don't have to wait for all the plugin developers to catch up with your update!

As a new feature, I would like to add custom fields on checkout forms!

**Comment by opr** [comment]
Hi @andreagaspari thanks for opening this discussion. I am glad to hear that you want to switch to the new Checkout Block, but also understand the frustration you are feeling about being blocked by incompatible plugins.

The new Cart and Checkout blocks were designed from the ground-up to solve many of the problems of the old checkout flow. Part of solving this problem involved re-assessing how we approach extensibility. Re-adding the mechanisms these plugins use to extend the checkout flow would likely result in the same issues re-surfacing.

If you are facing issues with a plugin then it'd be best reaching out to the developers of that plugin directly to ask them to consider making it work with the Cart and Checkout blocks.

By the way, I believe Mailchimp _is_ compatible with the new blocks?

> As a new feature, I would like to add custom fields on checkout forms!

We understand this feature has been widely requested, and so we worked on the [Additional Checkout Fields API](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md) - we're still working on bringing more field types to it, and also working on custom visibility rules for fields. More info can be found in this discussion: https://github.com/woocommerce/woocommerce/discussions/42995

I'll close this discussion now as I don't think there's anything actionable from us, but you'll still be able to reply.
